apps/charterafrica/src/components/Comments/Comment.js

```diff
@@ -0,0 +1,304 @@
+import RichTypography from "@/commons-ui/core/RichTypography";
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Avatar,
+  Box,
+  Button,
+  Chip,
+  IconButton,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  SvgIcon,
+  useTheme,
+} from "@mui/material";
+import PropTypes from "prop-types";
+import React, { forwardRef, useState, useRef } from "react";
+
+import chevronDown from "@/charterafrica/assets/icons/chevron-down, Size=24, Color=CurrentColor.svg";
+import chevronUp from "@/charterafrica/assets/icons/Chevron-up, Size=24, Color=CurrentColor.svg";
+import thumbsUp from "@/charterafrica/assets/icons/Type=thumbs-up, Size=24, Color=CurrentColor.svg";
+import userIcon from "@/charterafrica/assets/icons/Type=user, Size=auto, Color=CurrentColor.svg";
+import { neutral } from "@/charterafrica/colors";
+import LineClampedRichTypography from "@/charterafrica/components/LineClampedRichTypography";
+
+const DynamicLineClampedTypography = forwardRef(
+  ({ comment, ...props }, ref) => {
+    const ref1 = useRef();
+    const commentHeight = ref1?.current?.offsetHeight;
+    const currentTheme = useTheme();
+    const lineHeight =
+      currentTheme.typography.p3.lineHeight *
+      currentTheme.typography.p3.fontSize;
+    const showReadMore = commentHeight >= lineHeight * 3;
+
+    const [readMoreOpen, setReadMoreOpen] = useState(false);
+    const lineClamp = !readMoreOpen ? 2 : undefined;
+
+    return (
+      <>
+        <LineClampedRichTypography
+          ref={ref}
+          lineClamp={lineClamp}
+          variant="p3"
+          {...props}
+        >
+          {/* eslint-disable-next-line react/no-danger */}
+          <span dangerouslySetInnerHTML={{ __html: comment }} ref={ref1} />
+        </LineClampedRichTypography>
+
+        {showReadMore ? (
+          <Button
+            onClick={() => setReadMoreOpen((v) => !v)}
+            sx={(theme) => ({
+              ...theme.typography.p2,
+              mt: 2,
+              p: 0,
+              color: theme.palette.neutral.main,
+            })}
+          >
+            {readMoreOpen ? "Read less" : `Read more`}
+          </Button>
+        ) : null}
+      </>
+    );
+  }
+);
+const ThreadChildComment = forwardRef((props, ref) => {
```

Comment:
Shouldn't components like these be on their own files? And may be `CommentThread` is a better name.